Source/WebCore/platform/CaretAnimationSimple.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2022 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "CaretAnimationSimple.h"
+
+#include "RenderTheme.h"
+
+namespace WebCore {
+
+CaretAnimationSimple::CaretAnimationSimple(CaretAnimationClient& client, WeakPtr<Document, WeakPtrImplWithEventTargetData> document)
+    : CaretAnimation(Type::Simple, client, document)
+{
+}
+
+CaretAnimationSimple::~CaretAnimationSimple() = default;
+
+void CaretAnimationSimple::serviceAnimation(ReducedResolutionSeconds currentTime)
+{
+    if (isCaretBlinkingSuspended() && visible())
+        return;
+
+    const auto caretBlinkInterval = RenderTheme::singleton().caretBlinkInterval();
+
+    if (currentTime - m_lastTimeCaretPaintWasToggled >= caretBlinkInterval - (caretBlinkInterval * 0.01)) {
```

Comment:
this seems weird, but the delta time is always close to `caretBlinkInterval` but never exactly `caretBlinkInterval`. I'm not sure if there's a better way of handling this